File: examples/quickstart.ts
/**
 * Quickstart - Agentic Learning SDK (TypeScript)
 *
 * Get started with persistent memory in 30 seconds.
 *
 * Prerequisites:
 *     npm install @letta-ai/agentic-learning @anthropic-ai/sdk
 *     export ANTHROPIC_API_KEY="your-api-key"
 *     export LETTA_API_KEY="your-api-key"
 *
 * Usage:
 *     npx tsx quickstart.ts
 */

import Anthropic from "@anthropic-ai/sdk";
import { learning } from "@letta-ai/agentic-learning";

// Initialize LLM client
const client = new Anthropic();

async function askClaude(message: string) {
  console.log(`User: ${message}\n`);

  // That's it - wrap your API calls to enable persistent memory
  await learning({ agent: 'quickstart-ts-demo' }, async () => {
    const response = await client.messages.create({
      model: "claude-sonnet-4-20250514",
      max_tokens: 1024,
      messages: [{ role: "user", content: message }],
    });

    const text = response.content[0].type === "text" ? response.content[0].text : "";
    console.log(`Assistant: ${text}\n`);
  });
}

async function main() {
  // Memory automatically persists across LLM API calls
  await askClaude("My name is Bob and I love TypeScript.");

  await askClaude("What's my name and favorite language?");
}

main();

File: examples/quickstart.py
"""
Quickstart - Agentic Learning SDK

Get started with persistent memory in 30 seconds.

Prerequisites:
    pip install agentic-learning anthropic
    export ANTHROPIC_API_KEY="your-api-key"
    export LETTA_API_KEY="your-api-key"

Usage:
    python3 quickstart.py
"""

from anthropic import Anthropic
from agentic_learning import learning

# Initialize LLM client
client = Anthropic()

def ask_claude(message: str):
    """Send a message to Claude and print the response."""
    print(f"User: {message}\n")

    # That's it - wrap your API calls to enable persistent memory
    with learning(agent="quickstart-demo"):
        response = client.messages.create(
            model="claude-sonnet-4-20250514",
            max_tokens=1024,
            messages=[{"role": "user", "content": message}]
        )
        print(f"Assistant: {response.content[0].text}\n")

# Memory automatically persists across LLM API calls
ask_claude("My name is Alice and I love Python.")

ask_claude("What's my name and favorite language?")
